gutenberg.php

```diff
@@ -9,6 +9,8 @@
  * @package gutenberg
  */
 
+define( 'GUTENBERG__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
```

Comment:
These `define`s are duplicating code already in place elsewhere:  https://github.com/WordPress/gutenberg/blob/3f727da24ee9059759d4ab4669a6681839219a49/lib/client-assets.php#L13-L22

The `GUTENBERG__` (double underscore) convention is unique to the constants introduced here; others in the codebase do not use this convention.  It seems unnecessary to me.

I don't have a preference on whether this is done as a function or constants (the function is pretty fast), but it needs to be consistent in approach and naming.

Also, `gutenberg.php` should contain the bare minimum necessary to initialize the plugin, because it is going to be rewritten during the build process.  These defines should probably happen in a new `lib/constants.php` or similar.